src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurReportEmailServiceImpl.java

```diff
@@ -46,12 +47,42 @@ protected String buildEmailSubject(ConcurEmailableReportData reportData) {
         if(!reportData.getHeaderValidationErrors().isEmpty()) {
             sb.append("  There are header validation errors.");
         }
-        if(!reportData.getValidationErrorFileLines().isEmpty()) {
+        if(shouldAppendLineLevelValidationErrors(reportData.getValidationErrorFileLines())) {
             sb.append("  There are line level validation errors.");
         }
         return sb.toString();
     }
 
+    protected boolean shouldAppendLineLevelValidationErrors(List<ConcurBatchReportLineValidationErrorItem> lineValidationErrorItems) {
+
+        for(ConcurBatchReportLineValidationErrorItem concurBatchReportLineValidationErrorItem : lineValidationErrorItems) {
+            if(isValidLineLevelValidationError(concurBatchReportLineValidationErrorItem)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected boolean isValidLineLevelValidationError(ConcurBatchReportLineValidationErrorItem concurBatchReportLineValidationErrorItem) {
+
+        for (String errorMessage : concurBatchReportLineValidationErrorItem.getItemErrorResults()) {
+            if (errorMessage.contains("The line has the Pseudo (XXXX) payment code")) {
+                continue;
+            } else if (errorMessage.contains("The line detected as cash advance request. Should have been processed by first batch job step")){
+                continue;
+            } else if (errorMessage.contains("Cash Advance with key") && errorMessage.contains("did not have a corresponding entry from the Request Extract")) {
+                continue;
+            } else if (errorMessage.contains("Line was not processed because a Cash Advance under the same Report ID was missing a related Request Extract entry")) {
+                continue;
```

Comment:
Based on the requirement of only show the subject under certain conditions, I chose to go this route. Another solution could be to not add the error or add a flag based on the error. Let me know if you would suggest a different solution.

Reply:
I would not recommend this text-based analysis for determining the file error header logic, especially if the messages end up needing rewording in the future. I think a flag-based approach would be a much cleaner option, perhaps naming it something like `reportableAsLineLevelValidationError`. One of the main methods to be reworked is `ConcurStandardAccountingExtractCollectorBatchBuilder.reportUnprocessedLine()`, where you may need to add a third argument for the flag (and adjust the report item DTO's constructors accordingly). Since that method is also being used as a `BiConsumer` method reference, the relevant areas may need adjustment to use a three-arg reference.  (You may have to add a temporary 3-arg functional interface; Commons Lang3 does have `TriConsumer` interface but I think KFS is using a version that hasn't introduced that yet.)